src/features/Chat/components/ActiveWindow/Header/Buttons.tsx

```diff
@@ -42,39 +57,115 @@ const Buttons = ({ chat, showSearch }: Props) => {
-            />
+
+      {tabletMode ? (
+        <Container>
+          <IconButton
+            variant="search"
+            sizeInPx={24}
+            onClick={() => {
+              if (!isSearchShown) setIsTabletDropdownOpen(false);
+              setIsSearchShown(!isSearchShown);
+            }}
+          />
+          <IconButton
+            variant="menuLines"
+            sizeInPx={40}
+            onClick={() => {
+              if (!isTabletDropdownOpen) setIsSearchShown(false);
+              setIsTabletDropdownOpen(!isTabletDropdownOpen);
+            }}
+          />
+          {isSearchShown ? (
+            <TabletDropdown>
+              <Search hideSearch={hideSearch} />
+            </TabletDropdown>
+          ) : (
+            isTabletDropdownOpen && (
+              <Dropdown>
+                {chat.status === 'ok' ? (
+                  <>
+                    <TabletStatusButton
+                      onClick={() => {
+                        closeTabletDropdown();
+                        openDialog('archive');
+                      }}
+                      icon="archive"
+                      text={t('header.archive')}
+                    />
+                    <TabletStatusButton
+                      onClick={() => {
+                        closeTabletDropdown();
+                        openDialog('block');
+                      }}
+                      icon="block"
+                      text={t('header.block')}
+                    />
+                  </>
+                ) : (
+                  <TabletStatusButton
+                    onClick={() => {
+                      closeTabletDropdown();
+                      openDialog('restore');
+                    }}
+                    icon="return"
+                    text={t('header.restore')}
+                  />
+                )}
+                <TabletReportButton
+                  onClick={() => {
+                    closeTabletDropdown();
+                    openReportDialog();
+                  }}
+                  leftIcon={'danger'}
+                  sizeInPx={24}
+                  text={{
+                    color: 'purple',
+                    text: t('header.report'),
+                    variant: 'link',
+                  }}
+                />
+              </Dropdown>
+            )
+          )}
+        </Container>
+      ) : isSearchShown ? (
+        <Search hideSearch={hideSearch} />
+      ) : (
+        <Container>
+          <IconButton variant="search" sizeInPx={24} onClick={showSearch} />
+          {chat.status === 'ok' ? (
+            <>
+              <StatusButton
+                onClick={() => openDialog('archive')}
+                icon="archive"
+                text={t('header.archive')}
+              />
+              <StatusButton
+                onClick={() => openDialog('block')}
+                icon="block"
+                text={t('header.block')}
+              />
+            </>
```

Comment:
This should be refactored to multiplpe components. At least no nested ternaries minimum

Reply:
Fixed! 💪